src/Matcher/RecordedEventsThatMatcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace GQuemener\PhpSpec\Matcher;
+
+use PhpSpec\Matcher\BasicMatcher;
+use PhpSpec\Formatter\Presenter\Presenter;
+use PhpSpec\Exception\Example\FailureException;
+
+final class RecordedEventsThatMatcher extends BasicMatcher
+{
+    /**
+     * @var Presenter
+     */
+    private $presenter;
+
+    /**
+     * @param Presenter $presenter
+     */
+    public function __construct(Presenter $presenter)
+    {
+        $this->presenter = $presenter;
+    }
+
+    /**
+     * @param string $name
+     * @param mixed  $subject
+     * @param array  $arguments
+     *
+     * @return bool
+     */
+    public function supports(string $name, $subject, array $arguments): bool
+    {
+        return 'haveRecordedEventsThat' === $name
+            && 1 === \count($arguments)
+            && \is_callable($arguments[0])
+        ;
+    }
+
+    /**
+     * @param mixed $subject
+     * @param array $arguments
+     *
+     * @return bool
+     */
+    protected function matches($subject, array $arguments): bool
+    {
+        $refl = new \ReflectionObject($subject);
+        $prop = $refl->getProperty('recordedEvents');
+        $prop->setAccessible(true);
+        $events = $prop->getValue($subject);
```

Comment:
Using the aggregate translator would modify the aggregate root inner state (`$recordedEvents` would be reseted) which is not desirable here.